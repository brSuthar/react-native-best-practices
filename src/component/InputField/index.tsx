import React, {useState} from 'react';
import {TextInput} from 'react-native';

const InputField: React.FunctionComponent<IPropsInputField> = (
  props: IPropsInputField,
): React.JSX.Element => {
  const {value, placeholder, onChangeText} = props;
  return (
    <TextInput
      testID="input"
      value={value}
      placeholder={placeholder}
      onChangeText={onChangeText}
    />
  );
};

export default InputField;
