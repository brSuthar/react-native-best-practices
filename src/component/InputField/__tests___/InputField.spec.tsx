import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import 'react-test-renderer';
import InputField from '..';

describe('<InputField/>', () => {
  it('take snap for <INputField/>', () => {
    const field = render(<InputField value="" />);
    expect(field).toMatchSnapshot();
  });

  it('initial test', () => {
    const {getByTestId} = render(<InputField />);
    const input = getByTestId('input');
    expect(input.props.placeholder).toBeUndefined();
    expect(input.props.value).toBeUndefined();
    expect(input.props.onChangeText).toBeUndefined();
  });

  it('testing phase - one', () => {
    const {getByTestId} = render(
      <InputField
        value="phase 1"
        onChangeText={() => {}}
        placeholder="placeholder 1"
      />
    );
    const element = getByTestId('input');

    expect(element.props.value).toBe('phase 1');
    expect(element.props.placeholder).toBe('placeholder 1')
  });
});
