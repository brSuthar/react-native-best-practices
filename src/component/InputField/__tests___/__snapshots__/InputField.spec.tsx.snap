// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField/> take snap for <INputField/> 1`] = `
<TextInput
  testID="input"
  value=""
/>
`;
